lib/resque/worker.rb

```diff
@@ -619,6 +621,8 @@ def run_hook(name, *args)
     end
 
     def kill_background_threads
+      @heartbeat_thread ||= nil
```

Comment:
I like the pattern of having a reader method on these uninitialized instance variables (doing it in constructor is okay too). It makes it clearer what the default initialization value is. Is there a reason why we shouldn't do that for this variable?